### DISCUSSION TITLE
How to use subcommand argument modifiers like `txs:///foobar --xyz` to add option to latex command

### DISCUSSION THREAD

**Discussion by muzimuzhi** [texstudio-org/texstudio#3062] [Q&A] [answered]
For example, I want to use subcommand argument modifiers to add option `-shell-escape` to command `txs:///pdflatex`.

> | command | description |
> | --- | --- |
> | `txs:///foobar --xyz`        | This will add the `xyz` option |
> | `txs:///foobar/[--xyz=123]`  | This will change the value of the `xyz` option to `123` (i.e. removing any `xyz` option defined in `foobar`) |
> | `txs:///foobar/{--xyz=123}`  | This will remove `--xyz=123` from the `foobar` command line, ignoring `xyz` options with other values |
> | `txs:///foobar/{--xyz}`      | This will remove any `--xyz` option from the `foobar` command line, regardless of its value |
> | `txs:///foobar/{}`           | This will remove all options from the `foobar` command line, leaving only the name of the executable |
>
> _see doc https://texstudio-org.github.io/configuration.html#advanced-configuration-of-the-build-system with #3061 applied_

Using `txs:///pdflatex -shell-escape`, it expands to
```
pdflatex -synctex=1 -interaction=nonstopmode %.tex -shell-escape
```
The option `-shell-escape` is appended _after_ `%.tex`, hence ignored by program `pdflatex`.

I also tried with `txs:///pdflatex/{%.tex} -shell-escape %.tex`. It expands to
```
 pdflatex -synctex=1 -interaction=nonstopmode   -shell-escape
```
The `%.tex` is removed but not added again.

**Comment by TheComamba** [comment, marked as answer]
The version
`txs:///lualatex/{%.tex} -shell-escape "%.tex"`
works for me, although I think that this is more a hack than a feature.

**Reply by muzimuzhi** [reply]
OH ... good news anyway!

**Comment by muzimuzhi** [comment]
Note one should use `% !TeX TXS-program:compile = txs:///pdflatex<modifiers>"` instead of `% !TeX TS-program = txs:///pdflatex/<modifiers>`, see https://github.com/texstudio-org/texstudio/pull/3061#issuecomment-1500698027 and also https://github.com/texstudio-org/texstudio/discussions/3058#discussioncomment-5557488.
